addons/sale/models/sale_order_line.py

```diff
@@ -221,6 +221,10 @@ def write(self, values):
             precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
             self.filtered(
                 lambda r: r.state == 'sale' and float_compare(r.product_uom_qty, values['product_uom_qty'], precision_digits=precision) != 0)._update_line_quantity(values)
+            # Prevent computing the package type when we are setting the package quantity
+            if 'product_packaging_qty' in values and 'product_packaging_id' not in values:
+                # adding the packaging_id to the values prevents its recompute when the model is flushed
+                values['product_packaging_id'] = self.product_packaging_id.id
```

Comment:
Another problem here is that self could contain multiple records, making this line crash if the different lines have different packagings :confused:

Reply:
hmm, I guess you are right. Most of the time we are saving an SO, which does independent writes (with each its own set of `values`) for each SOL. But I've added the case when it's not the case, for example when explicitly calling `write` on multiple records of sol.